rx-lite/index.d.ts

```diff
@@ -232,7 +232,7 @@ declare namespace Rx {
         withLatestFrom<TOther, TResult>(souces: (Observable<TOther> | IPromise<TOther>)[], resultSelector: (firstValue: T, ...otherValues: TOther[]) => TResult): Observable<TResult>;
         concat(...sources: (Observable<T> | IPromise<T>)[]): Observable<T>;
         concat(sources: (Observable<T> | IPromise<T>)[]): Observable<T>;
-        concatAll(): Observable<T>;
+        concatAll(): T;
```

Comment:
```ts
const x: Observable<number> = Observable.range(0, 1);
x.concatAll().subscribe(x => console.log(x)); // throw `TypeError: xs.subscribe is not a function` at run time
```

This typescript code is valid now.  Current type definition is ill clearly.

I can't think of any better alternatives than this request.  Maybe, we should comment notice on code such as:

```ts
/**
 * Observable<Observable<V>>.concatAll(): Observable<V>
 * Notice: This method throws run time exception without Observable value.
 */
concatAll(): T;
```